### ISSUE TITLE
Uncategorized error for too many dots

### ISSUE THREAD

**Issue by electrum** [prestodb/presto#5596]
```
presto> select * from a.b.c.d;
Query 20160707_192115_20747_y6z74 failed: Too many dots in table name: a.b.c.d
java.lang.IllegalArgumentException: Too many dots in table name: a.b.c.d
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
	at com.facebook.presto.metadata.MetadataUtil.createQualifiedObjectName(MetadataUtil.java:119)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.visitTable(StatementAnalyzer.java:890)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.visitTable(StatementAnalyzer.java:219)
	at com.facebook.presto.sql.tree.Table.accept(Table.java:50)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:22)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.analyzeFrom(StatementAnalyzer.java:1733)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.visitQuerySpecification(StatementAnalyzer.java:1032)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.visitQuerySpecification(StatementAnalyzer.java:219)
	at com.facebook.presto.sql.tree.QuerySpecification.accept(QuerySpecification.java:125)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:22)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.visitQuery(StatementAnalyzer.java:829)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.visitQuery(StatementAnalyzer.java:219)
	at com.facebook.presto.sql.tree.Query.accept(Query.java:103)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:22)
	at com.facebook.presto.sql.analyzer.Analyzer.analyze(Analyzer.java:60)
	at com.facebook.presto.execution.SqlQueryExecution.doAnalyzeQuery(SqlQueryExecution.java:270)
	at com.facebook.presto.execution.SqlQueryExecution.analyzeQuery(SqlQueryExecution.java:256)
	at com.facebook.presto.execution.SqlQueryExecution.start(SqlQueryExecution.java:220)
	at com.facebook.presto.execution.QueuedExecution.lambda$start$1(QueuedExecution.java:68)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

**Comment by JackMc** [comment]
This has impacted a couple of our users trying to run queries where the table name is non-existent and 4 layers deep.

**Comment by electrum** [comment]
@JackMc This shouldn't happen if table does not exist. Do have an example query?

**Comment by JackMc** [comment]
Our query was `select * from hive.myschema.tablethatdoesntexist.something`. (with names replaced with non-internal names)

**Comment by cawallin** [comment]
Merged in #6055
